repose-aggregator/components/filters/body-extractor-to-header/src/main/scala/org/openrepose/filters/bodyextractortoheader/BodyExtractorToHeaderFilter.scala

```diff
@@ -0,0 +1,152 @@
+            if (is.markSupported()) is.mark(0)
+            val jsonString = Source.fromInputStream(is).mkString
+            if (is.markSupported()) is.reset()
+            logger.trace(s"is.available = ${is.available()}")
+            Try(JsonPath.using(jsonPathConfiguration).parse(jsonString))
+          } else {
+            new Failure(new Exception)
+          }
+        case _ =>
+          logger.debug("Content-Type was not defined.")
+          new Failure(new Exception)
+      }
+    }
+
+    extractions.foreach { extraction =>
+      if (jsonDoc.isSuccess) {
+        val extracted = jsonDoc match {
+          case Success(doc) => Try(doc.read[Any](extraction.jsonPath))
+        }
+
+        (extracted, extraction.defaultValue, extraction.nullValue) match {
+          // JSONPath value was extracted AND is NOT Null
+          case (Success(headerValue), _, _) if Option(headerValue).isDefined =>
+            addHeader(extraction.headerName, headerValue.toString, extraction.quality, extraction.overwrite)
+          // JSONPath value was extracted AND is Null AND NullValue is defined
+          case (Success(headerValue), _, Some(nullValue)) =>
+            addHeader(extraction.headerName, nullValue, extraction.quality, extraction.overwrite)
+          // JSONPath value was NOT extracted AND DefaultValue is defined
+          case (Failure(e), Some(defaultValue), _) =>
+            addHeader(extraction.headerName, defaultValue, extraction.quality, extraction.overwrite)
+          case (_, _, _) => // don't add a header
+        }
+      }
+    }
+
+    filterChain.doFilter(mutableHttpRequest, servletResponse)
+  }
+
+  override def destroy(): Unit = {
+    logger.trace("Body Extractor to Header filter destroying...")
+    configurationService.unsubscribeFrom(configurationFile, this.asInstanceOf[UpdateListener[_]])
+    logger.trace("Body Extractor to Header filter destroyed.")
+  }
+
+  override def configurationUpdated(config: BodyExtractorToHeaderConfig): Unit = {
+    extractions = config.getExtraction.asScala.map { extraction =>
+      Extraction(extraction.getHeader,
+        extraction.getJsonpath,
+        Option(extraction.getDefaultIfMiss),
+        Option(extraction.getDefaultIfNull),
+        extraction.isOverwrite,
+        Option(extraction.getQuality)
+      )
+    }
+    initialized = true
+  }
+
+  override def isInitialized: Boolean = initialized
+}
+
+object BodyExtractorToHeaderFilter {
+  private final val DEFAULT_CONFIG = "body-extractor-to-header.cfg.xml"
+  private final val SCHEMA_FILE_NAME = "/META-INF/schema/config/body-extractor-to-header.xsd"
+
+  case class Extraction(headerName: String,
+                        jsonPath: String,
+                        defaultValue: Option[String],
+                        nullValue: Option[String],
+                        overwrite: Boolean,
+                        quality: Option[java.lang.Double]
+                       )
```

Comment:
I think then Scala style guide suggests terminating the parameter list on the same line as the last parameter. See: 
http://docs.scala-lang.org/style/declarations.html#classes
http://docs.scala-lang.org/style/indentation.html#methods-with-numerous-arguments

Not a huge deal, but a thing we can totally do consistently.